### ISSUE TITLE
Magisk with libsu version after 4.0 show version as N/A 

### ISSUE THREAD

**Issue by nexplorer-3e** [topjohnwu/Magisk#7021]
Device & Android version: See #7020

Magisk version name: dd93556a
Magisk version code: 26102

followed instruction in #7014 with no luck.

[magisk_log_26102.log](https://github.com/topjohnwu/Magisk/files/11575347/magisk_log_26102.log)
[magisk_log_v24.1_ndebug.log](https://github.com/topjohnwu/Magisk/files/11575348/magisk_log_v24.1_ndebug.log)
[magisk_log_v25.0_ndebug.log](https://github.com/topjohnwu/Magisk/files/11575349/magisk_log_v25.0_ndebug.log)
![Screenshot_20230526-204251_Magisk](https://github.com/topjohnwu/Magisk/assets/60149113/6e5dc0ab-75d0-4b42-b76e-9e9fd13e32d4)

**Comment by nexplorer-3e** [comment]
> Duplicate of #6914

This ROM use https://github.com/html6405/android_device_samsung_n5100 as kernel, and there is no related logic that hide root.

**Comment by yujincheng08** [comment]
to confirm, when you launched magisk and it shows N/A, use a root file explorer or terminal to check if /system/bin/su exists?

**Comment by vvb2060** [comment]
Duplicate of #6976

**Comment by nexplorer-3e** [comment]
![Screenshot_20230527-022542_Termux](https://github.com/topjohnwu/Magisk/assets/60149113/8cad7d32-bb16-402b-8ff9-9308c1a451ad)
@vvb2060 From screenshot it can be indicated that magisk daemon is working, which differ from the case that described in XDA.


> to confirm, when you launched magisk and it shows N/A, use a root file explorer or terminal to check if /system/bin/su exists?

@yujincheng08 I remember that I invoke `su` from `/sbin` when systemless root starts. I would check my another device with magisk if there is a symlink in `/system/bin`
Doing diff on magisk logs, I figured out the log that exists in v25 (and 26102):
```logcat
W libbusybox.so: type=1400 audit(0.0:86): avc: denied { search } for name="block" dev=tmpfs ino=3104 scontext=u:r:untrusted_app:s0:c116,c256,c512,c768 tcontext=u:object_r:block_device:s0 tclass=dir permissive=0
W libbusybox.so: type=1400 audit(0.0:87): avc: denied { search } for name="block" dev=tmpfs ino=3104 scontext=u:r:untrusted_app:s0:c116,c256,c512,c768 tcontext=u:object_r:block_device:s0 tclass=dir permissive=0
```
So it may still be another weird ndk bug (or selinux issue), I may try replacing the new busy box with the old one to see whether it would get working after the day comes.

**Comment by nexplorer-3e** [comment]
[na.log](https://github.com/topjohnwu/Magisk/files/11582510/na.log)
[hang_splash.log](https://github.com/topjohnwu/Magisk/files/11582511/hang_splash.log)

tldr for the `git bisect log` output above: 
> first bad commit: [0ac19e3a4e99aa17c5bb7a7a791fc8f2114ddf1e] Fix app running without root <- this is the first commit that manager start to show N/A
> first bad commit: [f2c15c770135a5bb22c5820b65d09841ed608d87] Ensure RootService is launched <- this is the first commit that manager hangs at startup

Should try to bisect between v24.1 and v24.2 to see what breaks the second stage install.

**Comment by nexplorer-3e** [comment]
[abnormal_log.log](https://github.com/topjohnwu/Magisk/files/11583615/abnormal_log.log)

> # first bad commit: [71e37345b4d48febe5478dfd79287bbfc7042373] Update libsu

start from this commit, manager behave like below:

![Screenshot_20230528-113656_Magisk](https://github.com/topjohnwu/Magisk/assets/60149113/4050c538-751c-45af-a453-2494f7657eaa)

Kernel version: 3.0.101-lineage-g318c6b765916

**Comment by canyie** [comment]
Open Magisk app, then run `adb bugreport` and upload the produced bugreport file?

**Comment by nexplorer-3e** [comment]
> Open Magisk app, then run `adb bugreport` and upload the produced bugreport file?

[bugreport-konawifixx-PQ3A.190801.002-2023-06-09-16-07-00.zip](https://github.com/topjohnwu/Magisk/files/11703962/bugreport-konawifixx-PQ3A.190801.002-2023-06-09-16-07-00.zip)

Magisk version: 5e2ef1b7

**Comment by canyie** [comment]
```
06-09 16:06:52.623  root  1934  2169 D Magisk  : su: return code=[1]
06-09 16:06:52.636  root  1934  1936 D Magisk  : su: request from pid=[3647], client=[9]
06-09 16:06:52.636  root  1934  1936 D Magisk  : su: request from uid=[10170]
06-09 16:06:52.637  root  1934  1936 D Magisk  : su: waiting child pid=[3648]
06-09 16:06:52.637  root  3648  3648 D Magisk  : su: fork handler
06-09 16:06:52.638  root  3648  3648 D Magisk  : su: use namespace of pid=[3647]
06-09 16:06:53.503 10101  3422  3655 D ProfileInstaller: Skipping profile installation for com.steadfastinnovation.android.projectpapyrus
06-09 16:06:54.639 10119  3497  3657 D ProfileInstaller: Skipping profile installation for com.zacharee1.systemuituner
06-09 16:06:54.643  root  1934  1936 D Magisk  : su: return code=[1]
06-09 16:06:54.770 10170  3656  3656 W libbusybox.so: type=1400 audit(0.0:55): avc: denied { search } for name="block" dev=tmpfs ino=563 scontext=u:r:untrusted_app:s0:c170,c256,c512,c768 tcontext=u:object_r:block_device:s0 tclass=dir permissive=0
06-09 16:06:54.770 10170  3656  3656 W libbusybox.so: type=1400 audit(0.0:56): avc: denied { search } for name="block" dev=tmpfs ino=563 scontext=u:r:untrusted_app:s0:c170,c256,c512,c768 tcontext=u:object_r:block_device:s0 tclass=dir permissive=0
06-09 16:06:54.827 10170  3431  3611 W opjohnwu.magis: Long monitor contention with owner DefaultDispatcher-worker-1 (3610) at void java.lang.Object.wait(long, int)(Object.java:-2) waiters=0 in com.topjohnwu.superuser.internal.ShellImpl com.topjohnwu.superuser.internal.MainShell.get() for 3.154s
```
🤔 It's weird. I'll send you a version with more logs to test later

**Comment by canyie** [comment]
[app-debug.zip](https://github.com/topjohnwu/Magisk/files/11765613/app-debug.zip)
Try this build? Just install the app and send me the log saved from magisk app.

~~(off topic: are you also transgender? I'm a MTF :)~~

**Comment by nexplorer-3e** [comment]
> [app-debug.zip](https://github.com/topjohnwu/Magisk/files/11765613/app-debug.zip) Try this build? Just install the app and send me the log saved from magisk app.

[magisk_log_2023-06-12T22.59.51.log](https://github.com/topjohnwu/Magisk/files/11771112/magisk_log_2023-06-12T22.59.51.log)

**Comment by canyie** [comment]
```
06-12 22:59:18.183  5274  5302 D BUILDER : exec su --mount-master
06-12 22:59:28.226  5274  5302 D LIBSU   : java.io.IOException: Shell check timeout
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.ShellImpl.<init>(ShellImpl.java:116)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.BuilderImpl.build(BuilderImpl.java:148)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.BuilderImpl.build(BuilderImpl.java:136)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.BuilderImpl.build(BuilderImpl.java:103)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.MainShell.get(MainShell.java:50)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.MainShell.lambda$get$1(MainShell.java:71)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.MainShell$$ExternalSyntheticLambda1.run(Unknown Source:4)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.magisk.core.utils.DispatcherExecutor$execute$1.invokeSuspend(DispatcherExecutor.kt:24)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
06-12 22:59:28.226  5274  5302 D LIBSU   : Caused by: java.util.concurrent.TimeoutException
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at java.util.concurrent.FutureTask.get(FutureTask.java:206)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	at com.topjohnwu.superuser.internal.ShellImpl.<init>(ShellImpl.java:107)
06-12 22:59:28.226  5274  5302 D LIBSU   : 	... 15 more
```

**Comment by canyie** [comment]
[app-debug.zip](https://github.com/topjohnwu/Magisk/files/11880095/app-debug.zip)
Try this build and see if it works?

**Comment by nexplorer-3e** [comment]
> [app-debug.zip](https://github.com/topjohnwu/Magisk/files/11880095/app-debug.zip) Sorry for the long delay... Try this build and see if it works?

not working.
Installation via manager outputs error:
```log
! Process Error
! Installation Failed
```
via recovery it gave `Unsupported Image format` error. I flashed the older magisk patched boot and there are no logs (same as the normal debug variant, output some reflection warning)

**Comment by canyie** [comment]
Maybe some other regressions I think? Does this build still show version N/A?

**Comment by nexplorer-3e** [comment]
> Maybe some other regressions I think? Does this build still show version N/A?

Yes.

**Comment by canyie** [comment]
Try artifacts from the latest CI build? https://github.com/topjohnwu/Magisk/actions/runs/5617639740

**Comment by nexplorer-3e** [comment]
> Try artifacts from the latest CI build? https://github.com/topjohnwu/Magisk/actions/runs/5617639740

Works. Thank you!

**Comment by nexplorer-3e** [comment]
Another thing that I noticed is that the magisk cannot install by patch image in manager (saying `unknown image format`) **when no root**, it can only be installed via recovery. As the method is deprecated, more regression test is needed.

**Comment by canyie** [comment]
Yeah, I noticed some regressions as well, and I submitted some fixes to the master branch. Retry with the latest CI build? https://github.com/topjohnwu/Magisk/actions/runs/5625739626
